README.md

```diff
@@ -110,50 +212,19 @@ with a configuration like this:
 }
 ```
 
-Please note that because you currently need to turn `auto_https` support off, it
-is not advised to use the same instance of Caddy for your external-facing apps
-as you use for your internal-facing apps. This deficiency will be resolved as
-soon as possible.
-
-### Authenticating to the Tailcale network
+Please note that because you currently need to turn `auto_https` support off,
+you may want to run a separate Caddy instance for sites that do need `auto_https`.
 
-New nodes can be added to your Tailscale network by providing an [Auth
-key](https://tailscale.com/kb/1085/auth-keys/) or by following a special URL.
-Auth keys are provided to Caddy via the `TS_AUTHKEY` or `TS_AUTHKEY_<HOST>`
-environment variable. So if your network listener was `tailscale/myhost`, then
-it would look first for the `TS_AUTHKEY_MYHOST` environment variable, then
-`TS_AUTHKEY`.
-
-If no auth key is provided, then Tailscale will generate a URL that can be used
-to add the new node and print it to the Caddy log.
-Tailscale logs can be somewhat noisy so are turned off by default.
-Set the log level for the tailscale module to DEBUG to see the URL with [a Caddy logger](https://caddyserver.com/docs/caddyfile/options#log) like this:
-
-```yaml
-{
-  log tailscale {
-    output stdout
-    level DEBUG
-    include tailscale
-  }
-}
-```
-After the node had been added to your network, you can restart Caddy without the custom debug logger for tailscale.
-
-## Caddy authentication provider
+## Authentication provider
 
 Setup the Tailscale authentication provider with `tailscale_auth` directive.
-The provider will enforce that all requests are coming from a Tailscale user, as
-well as set various fields on the Caddy user object that can be passed to
-applications, similar to [nginx-auth][].
+The provider will enforce that all requests are coming from a Tailscale user,
```

Comment:
ah, good point!  Yes, this auth provider is largely redundant in terms of restricting access.  If you're only listening on the tailscale interface, then tailscale will enforce that only authorized users are able to reach the site.

Where the auth provider helps is when you want to pass information about the tailscale user to a backend application.  For example, I have a gitea instance that [is configured to use](https://docs.gitea.com/usage/authentication#reverse-proxy) `X-WEBAUTH-USER` to identify the logged in user.  So then I have my caddy config that is something like:

```
:80 {
        bind tailscale/gitea
        tailscale_auth
        reverse_proxy http://10.0.1.1:3000 {
                header_up X-Webauth-User {http.auth.user.tailscale_login}
        }
}
```

So then when I access http://gitea/, I'm automatically logged in based on my tailscale login name.  Grafana [supports the same thing](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/), which we use at Tailscale on our corporate tailnet.

The docs do talk about populating headers below, but I'll add an example of passing those in a reverse proxy.  As well as maybe clarify this specific sentence you're calling out, because the redundancy is indeed a little confusing.

Reply:
oh... I guess another example could be if you have a site that is both served over tailscale and elsewhere, for example on the public internet through Funnel or your own Funnel-like proxy.  You may want most of the site to be publicly accessible, but the admin section to require tailscale auth.  In that case, you could configure the auth provider just on `/admin/` paths, and then do the same thing about mapping tailscale user information.

Reply:
TIL you can do this with tailscale. That's actually very useful.